XdgThumbnailer/main.c

```diff
@@ -0,0 +1,139 @@
+#define G_LOG_DOMAIN "sameboy-thumbnailer"
+
+#include <gio/gio.h>
+#include <glib-object.h>
+#include <glib-unix.h>
+#include <glib.h>
+#include <signal.h>
+#include <stdlib.h>
+#include <string.h>
+
+// Auto-generated via `gdbus-codegen` from `interface.xml`.
+#include "build/obj/XdgThumbnailer/interface.h"
+
+static char const *const name_on_bus = "com.github.liji32.sameboy.XdgThumbnailer";
```

Comment:
String constants should be defined as `const char[]` do avoid the extra pointer.